hack/cloud-image-uploader/azure.go

```diff
@@ -0,0 +1,420 @@
+	StorageAccountType   string `json:"storageAccountType"`
+}
+
+// Mapping CPU architectures to Azure architectures
+var azureArchitectures = map[string]string{
+	"amd64": "x64",
+	"arm64": "arm64",
+}
+
+// AzureUploader represents an object that has the capability to upload to Azure
+type AzureUploader struct {
+	Options Options
+}
+
+func (azu *AzureUploader) AzureGalleryUpload(ctx context.Context) error {
+	var err error
+
+	var g *errgroup.Group
+	g, ctx = errgroup.WithContext(ctx)
+
+	fmt.Println("Setting default creds")
+
+	err = setDefaultAzureCreds()
+	if err != nil {
+		log.Fatalf("Error: %v\n", err)
+	}
+
+	fmt.Println("Getting locations")
+
+	err = getAzureLocations(ctx)
+	if err != nil {
+		log.Fatalf("Error: %v\n", err)
+	}
+
+	// Upload blob
+	fmt.Printf("Architectures: %+v\n", azu.Options.Architectures)
+
+	for _, arch := range azu.Options.Architectures {
+		arch := arch
+
+		g.Go(func() error {
+			fmt.Printf("Starting upload blob for %s\n", arch)
+			err = azu.uploadAzureBlob(ctx, arch)
+			if err != nil {
+				return fmt.Errorf("error uploading page blob for %s: %w", arch, err)
+			}
+			fmt.Printf("Starting disk creation for %s\n", arch)
+			err = azu.createAzureDisk(ctx, "azure-disk-template.json", arch)
+			if err != nil {
+				fmt.Printf("Error: %v\n", err)
+			}
+			fmt.Printf("Starting image version creation for %s\n", arch)
+			err = azu.createAzureImageVersion(ctx, "azure-image-version-template.json", arch)
+			if err != nil {
+				fmt.Printf("Error: %v\n", err)
+			}
+			return err
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		fmt.Println(err)
+	}
+
+	return nil
+}
+
+func (azu *AzureUploader) uploadAzureBlob(ctx context.Context, arch string) error {
+
+	var g *errgroup.Group
+	g, ctx = errgroup.WithContext(ctx)
+
+	blobURL := fmt.Sprintf("https://siderogallery.blob.core.windows.net/images/talos/talos-%s-%s.vhd", arch, azu.Options.Tag)
+
+	pageBlobClient, err := pageblob.NewClient(blobURL, cred, nil)
+	if err != nil {
+		log.Printf("Error: %v\n", err)
+	}
+
+	source, err := os.Open(azu.Options.AzureImage(arch))
+	if err != nil {
+		return err
+	}
+
+	defer func(source *os.File) {
+		err := source.Close()
+		if err != nil {
+
+		}
+	}(source) //nolint:errcheck
+
+	file, err := ExtractFileFromTarGz("disk.vhd", source)
+	if err != nil {
+		return err
+	}
+
+	data, err := io.ReadAll(file)
+	if err != nil {
+		log.Printf("Error: %v\n", err)
+		return err
+	}
+
+	reader := bytes.NewReader(data)
+	totalSize := int64(reader.Len())
+
+	// Check if the file size is a multiple of 512 bytes
+	if totalSize%pageblob.PageBytes != 0 {
+		panic("The file size must be a multiple of 512 bytes")
+	}
+
+	_, err = pageBlobClient.Create(ctx, totalSize, nil)
+	if err != nil {
+		log.Printf("Error: %v\n", err)
+	}
+
+	// Maximum size of the request body for a single UploadPages call is 4 MiB
+	chunkSize := 4 * 1024 * 1024
+	chunks := int(math.Ceil(float64(totalSize) / float64(chunkSize)))
```

Comment:
```suggestion
	chunks := int((totalSize + chunkSize -1)/chunkSize)
```

`chunkSize` should be a `const`